torchvision/datasets/triplet.py

```diff
@@ -0,0 +1,86 @@
+import torch
+import torch.utils.data as data
+
+import math
+
+
+def get_class_samples(num_classes, samples):
+    """Bins samples with respect to classes
+
+    Args:
+        num_classes (int): number of classes
+        samples (list[tuple]): (sample, class_idx) tuples
+
+    Returns:
+        list[list]: Bins of sample paths, binned by class_idx
+    """
+    class_samples = [[] for _ in range(num_classes)]
+    for sample_path, class_idx in samples:
+        class_samples[class_idx].append(sample_path)
+
+    return class_samples
+
+
+def generate_triplet(class_samples):
+    """Generates a triplet from bins of samples
+
+    Args:
+        class_samples (list[list]): bins of samples, binned by class
+        num_triplets (int): number of triplets to be generated
+
+    Returns:
+        list[tuple]: triplets of the form (anchor, positive, negative)
+    """
+    pos_cls, neg_cls = torch.multinomial(torch.ones(len(class_samples)), 2).tolist()
+    pos_samples, neg_samples = class_samples[pos_cls], class_samples[neg_cls]
+
+    anc_idx, pos_idx = torch.multinomial(torch.ones(len(pos_samples)), 2).tolist()
+    neg_idx = torch.multinomial(torch.ones(len(neg_samples)), 1).item()
+
+    return (pos_samples[anc_idx], pos_samples[pos_idx], neg_samples[neg_idx])
+
+
+class TripletDataset(data.IterableDataset):
+    """
+    A dataset with samples of the form (anchor, positive, negative), where anchor and
+    positive are samples of the same class, and negative is a sample of another class.
+    TripletDataset reads fram Dataset `dset` where `dset[i]` returns (sample_path, class_idx).
+
+    Args:
+        dset (Dataset): Dataset object where __getitem__ returns (sample_path, class_idx) tuple.
+        loader (callable): A function to load a sample given its path.
+        num_triplets (int): Number of triplets to generate before raising StopIteration.
+        num_classes (int): Number of classes in dset.
+        transform (callable, optional): A function/transform that takes in
+            a sample and returns a transformed version.
+            E.g, ``transforms.RandomCrop`` for images.
+
+    Attributes:
+        samples (list[tuple]): List of (anchor, positive, negative) triplets
+    """
+
+    def __init__(self, dset, loader, num_triplets, num_classes, transform=None):
```

Comment:
I think that the `num_classes` is not necessary, from my comment above

Reply:
Done.